### PR TITLE
Use GraphQL to query for PR labels

### PR DESCRIPTION
**Pull request by JasonEtco**
Chatting with @MylesBorins, it sounds like this query loop can hit rate limits. This PR swaps it to use `@octokit/graphql` and make one "large" query instead. This leverages GraphQL aliases to build a query string like this:

```graphql
query {
  repository (owner: "nodejs", name: "node") {
    pr1: pullRequest (number: 1) {
      labels (first: 100) {
        nodes {
          name
        }
      }
    }

    pr2: pullRequest (number: 2) {
      labels (first: 100) {
        nodes {
          name
        }
      }
    }

    pr3: pullRequest (number: 3) {
      labels (first: 100) {
        nodes {
          name
        }
      }
    }
  }
}
```

This'll return an object of `repository.pr1`, `repository.pr2`, etc. I've tried this GraphQL query with 5000 PRs at a time with no issues, and done some diffing against the current output vs. this PR's output with no changes 🎉 

I'd definitely suggest doing some more tests though, to verify that there are no output changes. @MylesBorins just let me know how I can help with that 👍

### CODE REVIEW

File: collect-commit-labels.js [hunk 1]
```diff
@@ -19,47 +18,43 @@ function collectCommitLabels (list, callback) {
...
       return callback(err)
     }
 
-    const q = async.queue((commit, next) => {
-      function onFetch (err, issue) {
-        if (err) {
-          console.error('Error fetching issue #%s: %s', commit.ghIssue, err.message)
-          return next()
-        }
+    const allPrs = list.map(item => item.ghIssue)
```
Comment (MylesBorins) [inline collect-commit-labels.js]: this is going to cause issues because you are using the full list and not the sublist.

Comment (MylesBorins) [reply collect-commit-labels.js]: (Sublist removes commits without meta data)

File: collect-commit-labels.js [hunk 2]
```diff
@@ -19,47 +18,43 @@ function collectCommitLabels (list, callback) {
...
+    const query = `query ($owner: String!, $name: String!) {
+      repository (owner: $owner, name: $name) {
+        ${bits}
       }
+    }`
 
-      if (commit.ghUser === 'iojs') {
-        commit.ghUser = 'nodejs' // forcibly rewrite as the GH API doesn't do it for us
+    graphql(query, {
+      owner: list[0].ghUser,
```
Comment (MylesBorins) [inline collect-commit-labels.js]: you similarly want to use sublist here instead of list

**Comment by MylesBorins** [comment]
hey @JasonEtco what's the status on this. Able to pick it up again?

**Comment by JasonEtco** [comment]
👋 Just getting back from a no-work vacation 🌴 Will pick it back up shortly. I need to rethink it a bit, since getting the correct owner/repo doesn't mesh super well with the single API request. I'll need to rework it to either be one GraphQL request per owner/repo, or have them all in one query segmented by owner/repo like:

```graphql
query {
  nodejs-node: repository (owner: "nodejs", name: "node") {
  
  }


  probot-proboot: repository (owner: "probot", name: "probot") {
  
  }
}
```

**Comment by rvagg** [comment]
This PR is out of date and needs a bit of work. I'm going to close it for now but https://github.com/nodejs/changelog-maker/pull/130 introduces octokit and GraphQL query for another purpose so it wouldn't be hard to use the same pattern there and the query from here to collect labels.

Up for graphs if someone wants to do this.

**Pull request closed without merging**